util/argo/argo.go

```diff
@@ -266,12 +239,110 @@ func ValidateRepo(
 	if err != nil {
 		return nil, fmt.Errorf("error getting enabled source types: %w", err)
 	}
+
+	sourceCondition := make([]argoappv1.ApplicationCondition, 0)
+	if spec.Sources != nil {
```

Comment:
Should this also check that len(sources) > 0?

Reply:
No more relevant. Replaced it with `spec.HasMultipleSources()`